src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,192 @@
+ordinal patterns
+```math
+C_{JS}[P] = Q_J[P, P_e] H_S[P]
+```
+where ``Q_J`` is originally the normalized Jensen-Shannon divergence of the ordinal distribution ``P`` to 
+a uniform distribution ``P_e`` with the same numer of bins, and ``H_S`` is the normalized permutation entropy, 
+`entropy_normalized(x, SymbolicPermutation())`.
+In this implementation, the distance measure defaults to the Jensen-Shannon divergence, but other (semi-)metrics can also be
+chosen. The type of entropy is also a free parameter, but defaults to the (originally used) Renyi entropy.
+
+For an ordinal distribution with a given permutation entropy, there is a minimum and maximum value that the
+statistical complexity can take. The minimum and maximum statistical complexity as a function of the entropy
+is calculated using [`minimum_complexity_entropy`](@ref) and [`maximum_complexity_entropy`](@ref).
+
+Standard usage:
+```julia
+m, τ = 3, 1
+est = SymbolicPermutation(; m, τ)
+c = StatisticalComplexity(; est, entropy=Renyi())
+x = randn(100)
+statistical_complexity = complexity(c, x)
+```
+
+NOTE: I'm not sure if this API is congruent with the general complexity API, but I think
+that, since the "standard" signature is also possible, it should be fine.
+I have implemented the other signature because I find it redundant to calculate the entropy
+twice, which is needed for the complexity anyway and usually calculated in combination with it.
+
+Because the distributions of permutation symbols are needed both for the permutation entropy and the
+statistical complexity, the same can also be achieved using
+```julia
+p = probabilities(est, x)
+permutation_entropy = entropy(est, x)
+statistical_complexity = complexity(c, p; h=permutation_entropy)
+```
+
+[^Rosso2007] Rosso, O. A., Larrondo, H. A., Martin, M. T., Plastino, A., & Fuentes, M. A. (2007). 
+            [Distinguishing noise from chaos](https://doi.org/10.1103/PhysRevLett.99.154102).
+            Physical review letters, 99(15), 154102. 
+"""
+Base.@kwdef struct StatisticalComplexity <: ComplexityMeasure
+    est::ProbabilitiesEstimator = SymbolicPermutation()
+    distance::SemiMetric = JSDivergence()
+    entropy::Entropy = Renyi()
+end
+
+alphabet_length(c::StatisticalComplexity) = alphabet_length(c.est)
+
+function complexity(c::StatisticalComplexity, p::Probabilities; h::Float64) 
+    L = alphabet_length(c)
+    # fill with the probability distribution zeros for empty bins, 
+    # we need this to compare to a uniform distribution of m! patterns
+    p = Probabilities(cat(zeros(L-length(p)), p.p, dims=1))
+    # normalization is done automatically by Probabilities
+    uniform = Probabilities(ones(L))
+
+    dist = evaluate(c.distance, p, uniform)
+    dist *= h
+
+    return dist
+end
+
+function complexity(c::StatisticalComplexity, x::AbstractArray)
+    p = probabilities(x, c.est)
+    L = alphabet_length(c)
+    h = entropy(c.entropy, p) / log(c.entropy.base, L)
```

Comment:
Because `entropy_normalized` is currently not implemented for taking a probability and I didn't like the idea of assuming the alphabet length from the number of bins, because empty bins could have been ommited (as far as I understood above, you didn't like this idea either).

I can of course implement a version that does this. The reason I want to do this in the first place is that since you need the probabilities both for the entropy and the complexity anyway, and the entropy itself is also needed for the complexity, I would like to calculate the probabilities only once.

Reply:
But you're right of course, I didn't consider that this doesn't work for every entropy, and I will change this of course.